File: Python/Algorithm/Job_Sequencing.py
# Complexity: O(n * log(n))

import heapq
def js(ar):
    ar.sort(); n, c, sum = len(ar), 0, 0; pq = []
    for i in range(n - 1, -1, -1):
        slot = ar[i][0] - (ar[i - 1][0] if i else 0)
        heapq.heappush(pq, -ar[i][1])
        while slot > 0 and pq:
            c += 1; sum += -heapq.heappop(pq); slot -= 1
    return c, sum

# Example operation
n = int(input()); ar = []
for _ in range(n):
    first, second = map(int, input().split())
    ar.append((first, second))
ans = js(ar)
print(ans[0], ans[1])

File: CPP/Algorithm/Job_Sequencing.cpp
// Complexity: O(n * log(n))

pair<int64_t, int64_t> aa[N]; int64_t n;
auto js() {
	sort(ar, ar + n); int64_t c = 0, sum = 0;
	multiset<int64_t, greater<>> st;
	for (int i = n - 1; i >= 0; --i) {
		int64_t slot = aa[i].first - (i ? aa[i - 1].first : 0);
		st.insert(aa[i].second);
		while (slot-- && !st.empty()) {
			++c; sum += *st.begin();
			st.erase(st.begin());
		}
	}
	return make_pair(c, sum);
}

// Operation:
	cin >> n;
	for (i = 0; i < n; ++i) cin >> aa[i].first >> aa[i].second;
	auto ans = js();
	cout << ans.first << ' ' << ans.second << '\n';